packages/@sanity/form-builder/src/inputs/PortableText/PortableTextInput.tsx

```diff
@@ -117,18 +117,43 @@ const PortableTextInputController = React.forwardRef(function PortableTextInputC
   const [ignoreValidationError, setIgnoreValidationError] = useState(false)
   const [invalidValue, setInvalidValue] = useState(null)
   const [isFullscreen, setIsFullscreen] = useState(false)
+  const [isActive, setIsActive] = useState(false)
+
+  // Set as active whenever we have focus inside the editor.
+  useEffect(() => {
+    if (hasFocus || focusPath.length > 1) {
+      setIsActive(true)
+    }
+  }, [hasFocus, focusPath])
+
+  const readOnly = useMemo(() => {
+    return isActive ? Boolean(readOnlyFromProps) : true
+  }, [isActive, readOnlyFromProps])
 
   const toast = useToast()
 
   // Memoized patch stream
-  const patches$: Subject<EditorPatch> = useMemo(() => new Subject(), [])
-  const patchObservable = useMemo(() => patches$.asObservable(), [patches$])
+  const patchSubject: Subject<{
+    patches: EditorPatch[]
+    snapshot: PortableTextBlock[] | undefined
+  }> = useMemo(() => new Subject(), [])
+  const remotePatch$ = useMemo(() => patchSubject.asObservable(), [patchSubject])
 
   const innerElementRef = useRef<HTMLDivElement | null>(null)
 
   const handleToggleFullscreen = useCallback(() => {
-    setIsFullscreen((v) => !v)
-    PortableTextEditor.focus(ref.current)
+    if (ref.current) {
+      const prevSel = PortableTextEditor.getSelection(ref.current)
+      setIsFullscreen((v) => !v)
+      setTimeout(() => {
```

Comment:
To help the next reader in understanding this better, could you explain in a comment why we have to delay the execution here?